### PR TITLE
Issue #29: Just splitting db server into another module

### PR DESCRIPTION
**Pull request by ekes**

### CODE REVIEW

**Comment by finnlewis** [comment]
Would be good to get some other eyes on this before merging.

**Comment by finnlewis** [comment]
@andybroomfield @stephen-cox @Adnan-cds it woudl be great to get some more eyes on this, especially on existing installed sites. 

This will free up enabling different backends.

**Comment by finnlewis** [comment]
This is linked to https://github.com/localgovdrupal/localgov_directories/pull/206

**Comment by finnlewis** [comment]
From discussion in the search working group: 

@ekes thinks it would be good to write an update path to enable the module but not install it's config! 

@ekes will need to look at this before further testing.

File: localgov_search.install [hunk 1]
```diff
@@ -16,3 +16,16 @@ function localgov_search_install() {
     localgov_search_entity_bundle_create('node', $bundle);
   }
 }
+
+/**
+ * Sitewide Search Search API database becomes its own module.
```
Comment (Adnan-cds) [inline localgov_search.install]: I am finding this description confusing :(

File: modules/localgov_search_db/localgov_search_db.install [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+function localgov_search_db_install($is_syncing) {
+  if (
+    !$is_syncing &&
+    ($index = Index::load('localgov_sitewide_search')) &&
+    ($index->getServerId() == '')
+  ) {
+    $server = Server::load('localgov_sitewide_search');
+    $module_path = \Drupal::service('extension.list.module')->getPath('localgov_search_db');
+    $index_solr_config = Yaml::decode(file_get_contents($module_path . '/config/conditional/search_api.index.localgov_sitewide_search.yml'));
+    $new_config_index = Index::create($index_solr_config);
```
Comment (Adnan-cds) [inline modules/localgov_search_db/localgov_search_db.install]: The variable name `$index_solr_config` looks out of context here.

**Review by Adnan-cds — APPROVED** [review]
Works for me and code looks mostly okay.  I have pointed out two minor issues.

I noticed that the search block is available on a fresh LocalGov site with the localgov_search module enabled and the localgov_search_db submodule disabled.  This is an unlikely scenario in real-life use.  So not a big deal but good to be aware of.

File: localgov_search.install (hunk 1 above)
Comment (ekes) [reply localgov_search.install]: Updated

File: modules/localgov_search_db/localgov_search_db.install (hunk 2 above)
Comment (ekes) [reply modules/localgov_search_db/localgov_search_db.install]: Corrected

**Comment by ekes** [comment]
> I noticed that the search block is available on a fresh LocalGov site with the localgov_search module enabled and the localgov_search_db submodule disabled.

What about https://github.com/localgovdrupal/localgov_search/pull/40/commits/32fd1d26ec35aec82361525cfeaae35ebdc4af52

**Comment by finnlewis** [comment]
Discussion in Merge Monday. 
To anyone with an existing site, this change should have no noticeable effect.
People with current fresh builds might need to change the config of their install process to include the new sub-module.
Adnan has reviewed and approved. 
Version number will bump the minor version to 1.1.0
So any objections to merging and doing a release?  None.

**Pull request merged**